### PR TITLE
Add fix for concurrency warning issue in slbeta2

### PR DESCRIPTION
**Pull request by kapilraajP**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

**Comment by LakshanSS** [comment]
This has to go to both branches

File: twitter/modules/listener/EventDispatcher.bal [hunk 1]
```diff
@@ -16,45 +16,32 @@
...
+        
+        string[] methodNames = adaptor.getServiceMethodNames();
+        self.isOnTweet = isMethodAvailable("onTweet", methodNames);
+        self.isOnReply = isMethodAvailable("onTweet", methodNames);
+        self.isOnReTweet = isMethodAvailable("onTweet", methodNames);
+        self.isOnQuoteTweet = isMethodAvailable("onTweet", methodNames);
+        self.isOnFollower = isMethodAvailable("onTweet", methodNames);
+        self.isOnFavourite = isMethodAvailable("onTweet", methodNames);
+        self.isOnDelete = isMethodAvailable("onTweet", methodNames);
+        self.isOnMention = isMethodAvailable("onTweet", methodNames);
```
Comment (Chuhaa) [inline twitter/modules/listener/EventDispatcher.bal:40]: "onTweet" is repeated in all the places.

File: twitter/tests/test.bal [hunk 2]
```diff
@@ -193,8 +193,10 @@ function testGetFollowers() {
     var userResponse = twitterClient->getFollowers(userID);
 
     if (userResponse is error) {
+        log:printInfo(userResponse.toString());
```
Comment (Chuhaa) [inline twitter/tests/test.bal:196]: We don't need to log info inside test cases.

File: twitter/modules/listener/EventDispatcher.bal [hunk 3]
```diff
@@ -16,45 +16,32 @@
...
+    private final boolean isOnTweet;
+    private final boolean isOnReply;
+    private final boolean isOnReTweet;
+    private final boolean isOnQuoteTweet;
+    private final boolean isOnFollower;
+    private final boolean isOnFavourite;
+    private final boolean isOnDelete;
+    private final boolean isOnMention;
+    private final HttpToTwitterAdaptor adaptor;
+    isolated function init(HttpToTwitterAdaptor adaptor) {
```
Comment (Chuhaa) [inline twitter/modules/listener/EventDispatcher.bal:29]:
```suggestion
    private final HttpToTwitterAdaptor adaptor;
    
    isolated function init(HttpToTwitterAdaptor adaptor) {
```
Can you add a new line

File: twitter/modules/listener/EventDispatcher.bal (hunk 1 above)
Comment (kapilraajP) [reply twitter/modules/listener/EventDispatcher.bal:40]: Fixed

File: twitter/modules/listener/EventDispatcher.bal (hunk 3 above)
Comment (kapilraajP) [reply twitter/modules/listener/EventDispatcher.bal:29]: Fixed

File: twitter/tests/test.bal (hunk 2 above)
Comment (kapilraajP) [reply twitter/tests/test.bal]: Removed those

**Review by LakshanSS — COMMENTED** [review]
Overall, looks good to me.

Please test locally and verify each listener event types are working.

**Review by abeykoon — APPROVED** [review]

**Pull request merged**